### PR TITLE
Substrate Updates

### PR DESCRIPTION
**Pull request by AurevoirXavier**
https://github.com/darwinia-network/substrate-update-tracking/issues/5
https://github.com/darwinia-network/substrate-update-tracking/issues/6

#506 
#511

### CODE REVIEW

**Comment by HackFisher** [comment]
We will not support sentry model anymore after latest changes, right?

Or removing sentry changes only apply to Pangolin node?

**Comment by AurevoirXavier** [comment]
> We will not support sentry model anymore after latest changes, right?
> 
> Or removing sentry changes only apply to Pangolin node?

Sentry mode was removed in upstream. I just apply the changes to our client.

**Comment by AurevoirXavier** [comment]
Ready for review.

But DO NOT merge into master.

File: bin/node/runtime/pangolin/src/lib.rs
Comment (HackFisher) [inline bin/node/runtime/pangolin/src/lib.rs:438]: The pallet ids are changed, will it have side effect or impacts unexpected?

It might be ok for Pangolin since it will be reset, but do we need to care the convention of compatibility on production networks (Crab and Darwinia).

File: frame/staking/src/lib.rs
Comment (HackFisher) [inline frame/staking/src/lib.rs:921]: Let's draft an issue as a reminder if we need to this sometime in future.

**Review by HackFisher — COMMENTED** [review]
What would be the dep problems since
`{ git = "https://github.com/darwinia-network/substrate.git", tag = "rococo-v1.1" }`
and 
`{ git = "https://github.com/paritytech/substrate.git", branch = "rococo-v1" }`

are pointing to the same commit 645299e ?

**Comment by AurevoirXavier** [comment]
> What would be the dep problems since
> `{ git = "https://github.com/darwinia-network/substrate.git", tag = "rococo-v1.1" }`
> and
> `{ git = "https://github.com/paritytech/substrate.git", branch = "rococo-v1" }`
> 
> are pointing to the same commit 645299e ?

Yes

File: bin/node/runtime/pangolin/src/lib.rs
Comment (AurevoirXavier) [reply bin/node/runtime/pangolin/src/lib.rs:438]: In Crab/Darwinia, we just need to increase the pallet index by 1.

**Comment by HackFisher** [comment]
The changes from polkadot-v0.8.29 to rococo v1 branch

https://github.com/paritytech/substrate/compare/polkadot-v0.8.29...rococo-v1

**Comment by AurevoirXavier** [comment]
Before merge, switch the deps back to darwinia-network/substrate

**Comment by HackFisher** [comment]
My question is for the darwinia-common used by next runtime, which substrate commit should we use.

To safely follow best upgrade practice,  it seems better to first upgrade substrate dep used by polkadot-v0.8.29.

But this might need to create a separate branch for merging, and which may share early changes, or we need to carefully check the changes from  polkadot-v0.8.29 branch to rococo v1 branch

**Comment by AurevoirXavier** [comment]
> My question is for the darwinia-common used by next runtime, which substrate commit should we use.
> 
> To safely follow best upgrade practice, it seems better to first upgrade substrate dep used by polkadot-v0.8.29.
> 
> But this might need to create a separate branch for merging, and which may share early changes, or we need to carefully check the changes from polkadot-v0.8.29 branch to rococo v1 branch

We should apply the tag 2.0.0 upgrade first.

---

And for this one.

1. Revert some changes before next runtime upgrade
2. Upgrade to rococo-v1 state
3. Maybe 0.8.30 is coming soon, check the migrations if there's not some big changes then skip 0.8.29

**Review by HackFisher — APPROVED** [review]

**Pull request merged**
